src/main/java/uk/nhs/digital/nhsconnect/nhais/model/mesh/MeshMessage.java

```diff
@@ -8,15 +8,39 @@
 @Getter @Setter @ToString @EqualsAndHashCode
 public class MeshMessage {
 
-    private String odsCode;
+    /**
+     * If SENDING TO MESH: set to the HA Trading Partner Code (recipient)
+     * If DOWNLOADING FROM MESH: DO NOT USE
+     */
+    private String haTradingPartnerCode;
+
+    /**
+     * If SENDING TO MESH: set based on the type of transaction being sent
+     * If DOWNLOADING FROM MESH: set to the Mex-WorkflowID response header
+     */
     private WorkflowId workflowId;
-    private String content;
+
     /**
-     * Correlation id associated with the request - used for distributed tracing
+     * If SENDING TO MESH: the content of the MESH message
+     * If DOWNLOADING FROM MESH: the body of the download response
      */
-    private String correlationId;
+    private String content;
+
     /**
-     * The timestamp (ISO format, UTC) when this message was sent - used for debugging and tracing
+     * If SENDING TO MESH: the time this message was published to the outbound mesh MQ
+     * If DOWNLOADING FROM MESH: the time this message was published to the inbound mesh MQ
      */
     private String messageSentTimestamp;
+
+    /**
+     * If SENDING TO MESH: DO NOT USE
+     * If DOWNLOADING FROM MESH: the message id of the message that was downloaded from MESH
+     */
+    private String meshMessageId;
+
+    /**
+     * If SENDING TO MESH: Correlation id for GP system
```

Comment:
"Value of OperationId response header for GP System" might describe it better

Reply:
Agree